client/me/purchases/payment-methods/payment-method-delete-dialog.tsx

```diff
@@ -1,34 +1,52 @@
-import { Button, Dialog } from '@automattic/components';
+import { Gridicon, Dialog } from '@automattic/components';
 import { useTranslate, TranslateResult } from 'i18n-calypso';
+import moment from 'moment';
 import { FunctionComponent } from 'react';
 import CardHeading from 'calypso/components/card-heading';
+import { withLocalizedMoment } from 'calypso/components/localized-moment';
+import { getPaymentMethodImageURL } from 'calypso/lib/checkout/payment-methods';
+import type { StoredPaymentMethod } from 'calypso/lib/checkout/payment-methods';
+import type { Purchase } from 'calypso/lib/purchases/types';
 
 import 'calypso/me/purchases/payment-methods/style.scss';
 
 interface Props {
+	card: StoredPaymentMethod;
 	paymentMethodSummary: TranslateResult;
 	isVisible: boolean;
 	onClose: () => void;
 	onConfirm: () => void;
+	moment: typeof moment;
```

Comment:
Since this is a functional component, we can use the `useLocalizedMoment` React Hook directly in the component rather than relying on the HOC. For example:

https://github.com/Automattic/wp-calypso/blob/d046df49527b632a2eb0c392b2dab6a8b63bbe8a/client/data/plugins/use-update-schedules-query.ts#L89

React Hooks generally provide better ergonomics than higher-order-components (there's a lot of reasons for this but it's probably not worth getting into here; I'm happy to explain later) and everything that can be done with a hook can be implemented as a HOC but an HOC can't easily be made into a hook.

Reply:
Ah, makes sense! Done. :)